x-pack/plugins/observability_solution/inventory/public/components/entities_grid/entity_name/index.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiLink, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React, { useCallback } from 'react';
+import {
+  AssetDetailsLocatorParams,
+  ASSET_DETAILS_LOCATOR_ID,
+  ServiceOverviewParams,
+} from '@kbn/observability-shared-plugin/common';
+import { useKibana } from '../../../hooks/use_kibana';
+import { EntityIcon } from '../../entity_icon';
+import { ENTITY_DISPLAY_NAME, ENTITY_TYPE } from '../../../../common/es_fields/entities';
+import { Entity, EntityType } from '../../../../common/entities';
+import { parseServiceParams } from '../../../utils/parse_service_params';
+
+interface EntityNameProps {
+  entity: Entity;
+}
+
+export function EntityName({ entity }: EntityNameProps) {
+  const { services } = useKibana();
+
+  const assetDetailsLocator =
+    services.share?.url.locators.get<AssetDetailsLocatorParams>(ASSET_DETAILS_LOCATOR_ID);
+
+  const serviceOverviewLocator =
+    services.share?.url.locators.get<ServiceOverviewParams>('serviceOverviewLocator');
+
+  const getEntityRedirectUrl = useCallback(() => {
+    const type = entity[ENTITY_TYPE] as EntityType;
```

Comment:
This manual typing is no longer needed.
```suggestion
    const type = entity[ENTITY_TYPE];
```